### PR TITLE
rewrote ScriptValidator

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
Ready

## Related Issues
fixes circle error from: https://circleci.com/gh/demisto/content/26838?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

## Description
while converting to a package, file validation raise BC error on docker image although it did not change.
rewrote the script validator to be like the integration validator.
when renaming (or removing and adding as new file), `git diff` return confusing changes

## Screenshots
docker image did not change (line 60)
image
yet, circle build failed:
image
`git diff` on the file returns `dockerimage` line as added:
image

## Related PRs
List related PRs against other branches:
https://github.com/demisto/content/pull/4043 - @erezh31 FYI

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [x] Code Review

### CODE REVIEW

**Comment by yaakovi** [comment]
the refactor fixed the problem as can be seen here:
https://circleci.com/gh/demisto/content/26994?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by yaakovi** [comment]
they are at the end of the `script_test` unit test file

**Comment by yaakovi** [comment]
@yuvalbenshalom i added another BC check to scripts (not specific for packagify)

File: Tests/scripts/hook_validations/script.py [hunk 1]
```diff
@@ -53,13 +53,50 @@ def _is_sub_set(cls, supposed_bigger_list, supposed_smaller_list):
...
+        current_args_to_required = self._get_arg_to_required_dict(self.current_script)
+        old_args_to_required = self._get_arg_to_required_dict(self.old_script)
+
+        for arg, required in current_args_to_required.items():
+            if required:
+                if (arg not in old_args_to_required) or \
+                        (arg in old_args_to_required and required != old_args_to_required[arg]):
+                    print_error("You've added required args in the script file '{}', the field is '{}'".format(
+                        self.file_path, arg))
+                    self._is_valid = False
```
Comment (yuvalbenshalom) [inline Tests/scripts/hook_validations/script.py]: please move the `self._is_valid = False` to the caller function that dose that anyway

**Pull request merged**